[AArch64][llvm] Armv9.7-A: Add support for SVE2p3 CVT operations

jthackray opened the pull request:
Add instructions for SVE2p3 CVT operations:
  - FCVTZSN
  - FCVTZUN
  - SCVTF
  - SCVTFLT
  - UCVTF
  - UCVTFLT

as documented here:

  * https://developer.arm.com/documentation/ddi0602/2025-09/
  * https://developer.arm.com/documentation/109697/2025_09/2025-Architecture-Extensions

jthackray committed:
[AArch64][llvm] Armv9.7-A: Add support for SVE2p3 CVT operations

Add instructions for SVE2p3 CVT operations:
  - FCVTZSN
  - FCVTZUN
  - SCVTF
  - SCVTFLT
  - UCVTF
  - UCVTFLT

as documented here:

  * https://developer.arm.com/documentation/ddi0602/2025-09/
  * https://developer.arm.com/documentation/109697/2025_09/2025-Architecture-Extensions

CarolineConcatto reviewed (commented):
Can we group the CVT tests into one file for the positive tests and one for the negative ones.

jthackray commented:
> Can we group the CVT tests into one file for the positive tests and one for the negative ones.

Sure, good idea, will do.

jthackray committed:
fixup! [AArch64][llvm] Armv9.7-A: Add support for SVE2p3 CVT operations

CR suggestions:
  - improve comments
  - combine positive and negative testcases into 2 files
  - add tests for fcvtzsn if Zd is a multiple of 2

jthackray commented on llvm/lib/Target/AArch64/SVEInstrFormats.td:
```diff
@@ -11300,3 +11300,49 @@ class sve_int_mla_cpa<string asm>
 }
+
+//===----------------------------------------------------------------------===//
+// FCVTZSN
```
Thanks, done.

jthackray committed:
fixup! [AArch64][llvm] Armv9.7-A: Add support for SVE2p3 CVT operations

Remove superfluous RUN lines

CarolineConcatto commented on llvm/test/MC/AArch64/SVE2p3/cvt-diagnostics.s:
```diff
@@ -0,0 +1,193 @@
+// Negative tests for instructions that are incompatible with movprfx
+
+movprfx z0, z7
+fcvtzun z0.b, { z0.h, z1.h }
```
Can we have 
z0.b, { z2.h, z3.h }

jthackray replied:
Sure, done.

CarolineConcatto commented on llvm/test/MC/AArch64/SVE2p3/cvt-diagnostics.s:
```diff
@@ -0,0 +1,193 @@
+// Negative tests for instructions that are incompatible with movprfx
+
+movprfx z0, z7
+scvtflt z0.h, z0.b
```
s/scvtflt z0.h, z0.b/scvtflt z0.h, z1.b/g

CarolineConcatto commented on llvm/test/MC/AArch64/SVE2p3/cvt-diagnostics.s:
```diff
@@ -0,0 +1,193 @@
+// Negative tests for instructions that are incompatible with movprfx
+
+movprfx z0, z7
+ucvtf z0.h, z0.b
```
ucvtf z0.h, z1.b

jthackray committed:
fixup! [AArch64][llvm] Armv9.7-A: Add support for SVE2p3 CVT operations

Adjust testcases for movprfx

jthackray commented on llvm/test/MC/AArch64/SVE2p3/cvt-diagnostics.s:
```diff
@@ -0,0 +1,193 @@
+// Negative tests for instructions that are incompatible with movprfx
+
+movprfx z0, z7
+scvtf z0.h, z0.b
```
Done.

CarolineConcatto reviewed (approved):
LGTM!
